Automattic/wp-calypso
# Edge Cache: Disable toggle if the site isn't public

Fixes https://github.com/Automattic/dotcom-forge/issues/2877

## Proposed Changes

Disables the edge cache toggle if the site isn't public. Also uses a more specific description if the site isn't edge cache eligible.

### Non-public

<img width="772" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/2e9b88cb-c04f-408b-aa9a-88f452698408">

### Public

<img width="758" alt="image" src="https://github.com/Automattic/wp-calypso/assets/36432/09341f35-1a2c-484e-89b5-3f607b2ae9d3">

## Testing Instructions

1. Create a new Business site and take it Atomic.
2. Navigate to Hosting Configuration.
3. Verify the edge cache toggle is disabled because the Business site hasn't been launched.
4. Open General Settings in a separate window.
5. Launch the site.
6. Refresh Hosting Configuration and verify the edge cache toggle can be activated.
7. Mark the site as Private.
8. Refresh Hosting Configuration and verify the edge cache toggle is disabled.
9. Mark the site as Coming Soon.
10. Refresh Hosting Configuration and verify the edge cache toggle is disabled.
11. Mark the site as Public.
12. Refresh Hosting Configuration and verify the edge cache toggle is enabled again.
